### PR TITLE
Refactor expense pool controller to use req.expensePool

### LINKED ISSUE #115: Refactor expense pool controller to use req.expensePool
Since authorizing expense pool membership decorates the request with the expense pool, the expense pool controller should be able to use that found instance instead of looking it up again.

### PR DESCRIPTION
**Pull request by dting**
Resolves #115

### CODE REVIEW

File: server/api/expense-pool/expense-pool.controller.js [hunk 1]
```diff
@@ -21,65 +21,30 @@ controller.addPool = (req, res) => {
   .catch(error => handleError(res, error));
 };
 
-
 controller.addUserToPool = (req, res) => {
```
Comment (hannahhenderson) [inline server/api/expense-pool/expense-pool.controller.js:24]: Why is ExpensePool commented out?

**Comment by hannahhenderson** [comment]
LGTM

File: server/api/expense-pool/expense-pool.controller.js (hunk 1 above)
Comment (dting) [inline server/api/expense-pool/expense-pool.controller.js:24]: Hmm I'm not sure which commented out ExpensePool you are referring to. 

If you mean why the look up for expense pool is removed. All of these routes dealing with a specific expense pool are behind the `isPoolMember` authorization middleware. The `req` will always have a `expensePool` property with the expense pool specified by the `req.params.id`. That means we don't need to look up the expense pool in these controller functions.

**Comment by hannahhenderson** [comment]
Thank you for resolving the merge conflicts!

**Pull request merged**